### ISSUE TITLE
Can't transfer save states to snes

### ISSUE THREAD

**Issue by ghost** [ClusterM/hakchi2#578]
I tried using the save state manager to move my mgba save state and save file to my snes but it didnt work. Is there some way to convert it or put the save file directly on the snes mini?

**Comment by fluffiebuddie** [comment]
Having the exact same problem here after trying 100 times..

**Comment by ClusterM** [comment]
Try to create save on SNES Mini, backup it, unpack at as .tar.gz archive, replace SRAM file, repack, upload to SNES Mini.

**Comment by ghost** [comment]
I tried to do that but the save/state didnt load. How do i get the sram file from the emulator save?

**Comment by jgoldsack** [comment]
the CLVS file is just a tar.gz file. Have to open it up there to replace the SRAM file. That said, I did that as well, and when I loaded up the game, the save state that was there previously was erased. There is a sram.hash file in there as well, I do not know how that comes into play yet though.

**Comment by fabiophillip** [comment]
same here! cannot load the state eve after replacing the SRAM file. I replaced it with my srm file because i Didn't have a SRAM file

**Comment by DanBonez** [comment]
For all the games, the string of hex in the sram.hash file gets added to the end of the cartridge.sram file. To import an .srm file, you will need to open it with a hex editor, do a checksum analysis for SHA-1, copy that string of hex values and add it to the end of the .srm file. Then save that file as cartridge.sram, delete the sram.hash file and import your save.

Confirmed that works for Earthbound, Final Fantasy II, Chrono Trigger, Super Mario World.

So far does not work for Super Mario RPG and Yoshi's Island. Can't seem to figure out why this isn't working yet for those two, haven't had any luck. Have not tried any other games.

**Comment by fabiophillip** [comment]
how to checksum analysis for SHA-1? is it hard? I'm a total noob at this, sorry ^^'

**Comment by DanBonez** [comment]
I used the hex editor HXD (easy to find and free to download). Open the .srm file with that, click "analysis" drop-down, then checksum, and select SHA-1. It will show a string of hex values in the bottom window, just copy that and paste it at the end after the last row.

**Comment by fabiophillip** [comment]
> copy that string of hex values and add it to the end of the .srm file. Then save that file as cartridge.sram, delete the sram.hash file and import your save.

Do i copy it in the same line as the last line? Do i delete de .hash file and then replace the .sram file with my file? or do I rename the file to .hash format?

**Comment by DanBonez** [comment]
You will paste it on the next line after the last line in the .srm file. Once you do that, save your .srm file as cartridge.sram and replace the original file with your new file. Make sure to delete the .hash file... it is not necessary and will automatically be recreated once you start a game once.

**Comment by fabiophillip** [comment]
> You will paste it on the next line after the last line in the .srm file. 
you mean right after the game name in the last line, right? I used HxD editor and it didn't give me the opportunity to create a new line, but i copied write (Ctrl + B) right after the last digit in the last line! Am I right?

**Comment by fabiophillip** [comment]
Thank you for keeping up trying to help me! I'm really grateful ^^
I created the sram file (applied checksum-1, copied the line result and pasted at the last line with a paste-write ctrl + B), 
exported the clvs file to my computer,
renamed the extension to .tar.gz, extracted with winrar, 
replaced the sram, 
deleted the .hash file, 
repacked it to the .tar.gz format, 
renamed the extension to clvs
then imported to my snes mini with save-state manager(without deleting the other save). did I do something wrong? because when I tried loading the game, it just had the save I first created in the snes mini, not the one I wanted it to be...

**Comment by fabiophillip** [comment]
can you tell me if I did it all right? please?^^

**Comment by DanBonez** [comment]
Looks like you did it right. What game are you attempting to do this with? I've been unable to get some games to work when I've tried (Super Mario RPG and Yoshi's Island).

**Comment by DanBonez** [comment]
[http://64.91.255.7/forum/showpost.php?s=184ab271e940707df9dacd5bd7698851&p=251151590&postcount=7921](url)

Here's a good link that has the steps broken down in pics if you want to double check.

**Comment by fabiophillip** [comment]
thanks, DanBonez but the link seems down ^^'

**Comment by fabiophillip** [comment]
I'm trying to transfer the dragon quest V fan-translated game

**Comment by DanBonez** [comment]
Tested DQV fan translation, works with the described method. It should look something like this: 

![image](https://user-images.githubusercontent.com/32676448/31517472-a392246a-af61-11e7-974e-58cd1b111725.png)
![image](https://user-images.githubusercontent.com/32676448/31517481-ac6e43de-af61-11e7-80e3-382286dec3da.png)


Another note, I've had better luck transferring the files via FTP than using the save state manager.

**Comment by fabiophillip** [comment]
really? great! then maybe it's something I'm doing wrong... If I've successfully swapped the .sram file from the emulator to the snes mini, yet snes mini wouldn't recognize it... then I think snes mini would show me a game without any saved data, right? And yet, yesterday, when I tried this method, I swapped the sram and snes mini showed me the saved data which I saved in the snes mini just for tests. This means maybe... Haxchi didn't import my save! It didn't even swap my save! I think... Do you delete the saved state in snes mini before you transfer the altered .clvs file? I tried, but then when I try to import it after deleting the saved state, the game name won't appear, so I think it doesn't work this way... do you have another site with the steps with pics? thank you so much for testing!^^

**Comment by fabiophillip** [comment]
how did you transfer the save using hakchi? is it with the save-state manager? you just click import and then import the clvs file?

**Comment by fabiophillip** [comment]
I connected to snes mini, but in what folder is my save?

**Comment by DanBonez** [comment]
var->lib->clover->profiles->0-> then the game ID

**Comment by fabiophillip** [comment]
One more time, really THANK YOU for keeping up with me! I'm really grateful!^^
I think I'm almost there, but missing something. here are my screenshots. I'm tring to import the save from my snes9x emulator in the PC. the save in the pc works fine, I've just tested it an saved with the nearest priest

![picture1](https://user-images.githubusercontent.com/4417273/31549574-00345d3e-b005-11e7-9730-dfe62f7f0433.png)
![picture2](https://user-images.githubusercontent.com/4417273/31549575-0052305c-b005-11e7-84c1-2b0cd5272d6b.png)
![picture3](https://user-images.githubusercontent.com/4417273/31549576-0075c724-b005-11e7-9cb2-a8bcebfc09c0.png)
![picture4](https://user-images.githubusercontent.com/4417273/31549577-0090f904-b005-11e7-83a7-49db096e3bfc.png)
![picture5](https://user-images.githubusercontent.com/4417273/31549579-00a9689a-b005-11e7-9989-db5cf6d42de0.png)
![picture6](https://user-images.githubusercontent.com/4417273/31549581-00d174ca-b005-11e7-9714-7ba2bd3f57a3.png)
![picture7](https://user-images.githubusercontent.com/4417273/31549583-00f91cf0-b005-11e7-963d-8e3b280e1cf5.png)
![picture8](https://user-images.githubusercontent.com/4417273/31549584-0113f1ec-b005-11e7-8387-4513a11940bc.png)
![picture9](https://user-images.githubusercontent.com/4417273/31549585-01611c9c-b005-11e7-9ba3-e61488e62946.png)

**Comment by DanBonez** [comment]
You are doing everything right per the screen shots, but the Hex string from your SHA-1 checksum is too long. It should only be 40 total characters long (20 hex entries). Looks like you are pasting it in the wrong area. The checksum should be pasted in the area with the hex entries, if that makes sense. 

https://www.reddit.com/r/miniSNESmods/comments/761m8j/srm_to_canoecloverclvs_converter/

On that note, someone made a tool that does the conversion automatically. I haven't tried it yet (still at work), but it might be worth a shot.

**Comment by fabiophillip** [comment]
thankyouthankyouthankyou it worked! Thank you so much for your help! I used this python converter and transferred the cartridge.sram via FTP and it worked!Many thanks!^^

**Comment by Drakonas** [comment]
Next time please ask this kind of question on the MiniSNESMods subreddit or the Nintendo Classics discord. This is a Hakchi bug report tracker, not a personal help line.

**Comment by Netfreak25** [comment]
Works for WiiU VC also (Tested TLOZ)

I exported from Wiiu using Savemii_mod
then opened the VES file in HxD, selected and copied all the bytes starting at offset 0x30 (starting at the third row) all the way to end, then pasted these bytes in a new document. Save the new document as a srm file (Now it should run on various emulators)​
At the end I added the SHA-1 checksum of the file to end of the file, like explained above.
Created an empty savegame on SNES mini, extracted it and replaced cartridge.sram with the just created file. Compressed, uploaded, done :)
